mlflow/models/model.py

```diff
@@ -186,7 +186,8 @@ def log(
         with TempDir() as tmp:
             local_path = tmp.path("model")
             run_id = mlflow.tracking.fluent._get_or_start_run().info.run_id
-            mlflow_model = cls(artifact_path=artifact_path, run_id=run_id)
+            model_uuid = uuid.uuid4().hex
+            mlflow_model = cls(artifact_path=artifact_path, run_id=run_id, model_uuid=model_uuid)
```

Comment:
We should not set uuid here. Otherwise there're many other place we need also set uuid. 
e.g. 
https://github.com/mlflow/mlflow/blob/f9b5745b68c815adb411453048317ee46ac52a31/mlflow/pyfunc/__init__.py#L1072
https://github.com/mlflow/mlflow/blob/f9b5745b68c815adb411453048317ee46ac52a31/mlflow/catboost.py#L122
etc.

I move the set uuid code into Model constructor again.